packages/reaction-core/server/methods/shop.js

```diff
@@ -363,6 +363,35 @@ Meteor.methods({
   },
 
   /**
+   * shop/createTag
+   * @summary creates new tag
+   * @param {String} tagName - new tag name
+   * @param {Boolean} isTopLevel - if true -- new tag will be created on top of
+   * tags tree
+   * @return {String} with created tag _id
+   */
+  "shop/createTag": function (tagName, isTopLevel) {
+    check(tagName, String);
+    check(isTopLevel, Boolean);
+
+    // must have 'core' permissions
+    if (!ReactionCore.hasPermission("core")) {
+      throw new Meteor.Error(403, "Access Denied");
+    }
+
+    // we assume what tag is not exist.
+    const tag = {
```

Comment:
I forgot to remove this comment from previous redaction. But you are right about simple check for tag existence and this check happens before this method calls. In other words, when we call it, we already know that there is no such tag.